### DISCUSSION TITLE
How to properly use  function call where a mapping is expected

### DISCUSSION THREAD

**Discussion by maru-sama** [python/typing#1418] [Q&A]
Hi, I guess this is a pretty simple question, but I cannot figure it out. I am using the httpx.get method and want to specify headers. My call (simplified) looks like this...

httpx.get("https://example.com", headers={"one": "two"})

Mypy is complaining with the following error...
Argument "headers" to "get" of "Client" has incompatible type "dict[str, str|None]"; expexted "Headers| Mapping[str, str,]| Mapping[bytes, bytes]|Sequence[tuple[str, str]]| Sequence[tuple[bytes,bytes]] | None

The code works and I can "fix" this issue by specifying a variable as Mapping[str,str] first and then using it, but I think this is a little bit too complex, is there a better solution for this?

**Comment by erictraut** [comment]
Are you sure you're running the latest version of mypy (1.4) and have the latest version of `httpx` (0.24.1) installed? Your code type checks without errors for me in both mypy and pyright.

Here's a minimal, self-contained code sample:

```python
import httpx
httpx.get("https://example.com", headers={"one": "two"})
```

**Comment by maru-sama** [comment]
Sorry, I should have shown my real code, but youre reply made me think....

This is working...

`
ser = cl.get(
  "https://api.twitch.tv/helix/users",
   headers={
   "Authorization": f"Bearer {self.config['access_token']}",
   "Client-Id": self.config["client_id"],
  },
)
`

but not this

`
ser = cl.get(
  "https://api.twitch.tv/helix/users",
   headers={
   "Authorization": f"Bearer {self.config.get('access_token')}",
   "Client-Id": self.config.get("client_id"),
  },
)
`
I guess the .get() call is creating an issue since it can return "None",

**Comment by maru-sama** [comment]
Okj, it is the last assignment, if I use self.config.get("client_id", "") it works. Sorry that I did not try this before, it makes a lot of sense that this throws an error.
